retrieval/discovery/discovery.go

```diff
@@ -41,12 +39,16 @@ func NewKubernetesDiscovery(conf *config.KubernetesSDConfig) (*kubernetes.Discov
 }
 
 // NewMarathon creates a new Marathon based discovery.
-func NewMarathon(conf *config.MarathonSDConfig) *marathon.Discovery {
-	return &marathon.Discovery{
-		Servers:         conf.Servers,
-		RefreshInterval: time.Duration(conf.RefreshInterval),
-		Client:          marathon.FetchApps,
+func NewMarathon(conf *config.MarathonSDConfig) (*marathon.Discovery, error) {
+	md := &marathon.Discovery{
+		Conf:       conf,
+		AppsClient: marathon.FetchApps,
+	}
+	err := md.Initialize()
```

Comment:
Is there an advantage of having a separate `Initialize` function instead of having a `NewDiscovery()` that directly returns a completely initialized struct? (like Consul/DNS)

It seems counterintuitive to have the initialization split over two functions in two different packages. I'd probably also get rid of `newMarathonHTTPClient` as a separate function as it is mostly struct initialization as well.